seaborn/categorical.py

```diff
@@ -1577,6 +1577,10 @@ def boxplot(
     fliersize=None, hue_norm=None, native_scale=False, log_scale=None, formatter=None,
     legend="auto", ax=None, **kwargs
 ):
+    
+    if "positions" in kwargs:
+        msg = "boxplot() does not support parameter 'positions'. Consider to use native_scale=True and specify positions via parameter 'x'."
+        raise ValueError(msg)
```

Comment:
Small thing but `TypeError` is probably a better error class here, that is what you get if you passed an undefined parameter. Alternatively, `RuntimeError` since `positions` is technically valid but produces incorrect behavior.